components/suggest/src/db.rs

```diff
@@ -141,6 +143,11 @@ impl<'a> SuggestDao<'a> {
                      FROM suggestions s
                      JOIN pocket_keywords k ON k.suggestion_id = s.id
                      WHERE k.keyword_prefix = :keyword_prefix
+                     UNION ALL
```

Comment:
This will query both amo and pocket keyword tables despite that the consumer might need one of them. I think we have two options here:
- Merge `amo_keywords` and `pocket_keywords` as they're almost identical with the same matching logic. We can split them up if they diverge in the future
- Split this SQL into multiple so that it will query suggestions per provider, which means it'd issue multiple calls against the database. The upside is that we don't need to maintain this SQL for different keyword tables

Reply:
makes sense, let me try to fiddle with both options

Reply:
okie doke, so i think i was able to implement the first option, currently trying the second